includes/Checker/Runtime_Environment_Setup.php

```diff
@@ -48,6 +48,10 @@ static function () use ( $permalink_structure ) {
 				}
 			);
 
+			if ( ! isset( $_SERVER['HTTP_HOST'] ) ) {
+				$_SERVER['HTTP_HOST'] = '';
+			}
```

Comment:
This is lacking an explanatory comment.

Why are we running into this problem though in the first place? It would be great if we could address the root cause rather than papering over it.

From what I saw last time, the notice came from `wp_guess_url()`. A stack trace could tell us why it's being called.

My _guess_ from looking at the WordPress source code is as follows though:

1. `Runtime_Environment_Setup::set_up()` calls `wp_install()` to set up a new database for the runtime check
2. `wp_install()` calls `populate_options()`, which in turn calls `wp_guess_url()`. It also calls `wp_guess_url()` itself to set the `siteurl` option.

So IMHO the better way to fix this is by setting `$_SERVER['HTTP_HOST']` in `Runtime_Environment_Setup::set_up()`. Not here.

We can set it to the existing `get_option( 'siteurl' )` value from the actual WordPress site.

What do you think?

Reply:
Yes. Also this would probably have other side effects also. I added this in a guess as I am not being able to reproduce this in my local setup or even local Behat test. I will update PR as per your feedback.

Reply:
This code is actually in the `Runtime_Environment_Setup::set_up()`. Is it meant to be kept somewhere else?

Reply:
Oh wow 🤦  I didn't realize that, sorry! For some reason I thought this was in `FeatureContext`.